server/nimble_server.py

```diff
@@ -19,22 +19,41 @@
 # Allow imports from the parent directory (potential security issue)
 sys.path.append(os.path.dirname(__file__) + "/..")
 
-from generate import generate_docs
+from generate import generate
 
 app = FastAPI()
 
 # Static files for the test page
 app.mount("/static", StaticFiles(directory="static"), name="static")
 
 
-# http://127.0.0.1:8000/wakoma/nimble/test_page
-@app.get("/wakoma/nimble/test_page")
+# http://127.0.0.1:8000/wakoma/nimble/configurator
+@app.get("/wakoma/nimble/configurator")
 async def read_index():
     """
     Loads a sample page so that the system can be tested end-to-end.
     """
 
-    return FileResponse('test_page.html')
+    return FileResponse('configurator.html')
+
+
+def get_unique_name(config):
+    """
+    Allows the caller to get a unique name for a configuration.
+    """
+    unique_name = ""
```

Comment:
`unique_name = "".join(config)`
could replace lines 44-48
`unique_name = "!".join(config)`
could be used to use ! as a delimiter. If wanted.

Reply:
The delimiter needs to be acceptable in filenames, as this helps make the filesystem caching mechanism work.

Reply:
`!` should be safe in windows and unix which is why i picked it over more conventional delimiters